packages/host/app/components/operator-mode/code-submode/boxel-spec-preview.gts

```diff
@@ -0,0 +1,274 @@
+
+import { LoadingIndicator } from '@cardstack/boxel-ui/components';
+
+import {
+  type ResolvedCodeRef,
+  catalogEntryRef,
+  getCards,
+  type Query,
+} from '@cardstack/runtime-common';
+
+import { type ModuleDeclaration } from '@cardstack/host/resources/module-contents';
+import OperatorModeStateService from '@cardstack/host/services/operator-mode-state-service';
+
+import RealmService from '@cardstack/host/services/realm';
+
+import { type CardDef } from 'https://cardstack.com/base/card-api';
+
+import { CatalogEntry } from 'https://cardstack.com/base/catalog-entry';
+
+import { type FileType } from '../create-file-modal';
+
+import type { WithBoundArgs } from '@glint/template';
+
+interface Signature {
+  Element: HTMLElement;
+  Args: {
+    selectedDeclaration?: ModuleDeclaration;
+    createFile: (
+      fileType: FileType,
+      definitionClass?: {
+        displayName: string;
+        ref: ResolvedCodeRef;
+      },
+      sourceInstance?: CardDef,
+    ) => Promise<void>;
+  };
+  Blocks: {
+    default: [
+      WithBoundArgs<typeof BoxelSpecPreviewTitle, 'showCreateBoxelSpecIntent'>,
+      WithBoundArgs<
+        typeof BoxelSpecPreviewContent,
+        | 'showCreateBoxelSpecIntent'
+        | 'boxelSpecInstances'
+        | 'selectedInstance'
+        | 'createBoxelSpec'
+        | 'selectBoxelSpec'
+      >,
+    ];
+  };
+}
+
+interface TitleSignature {
+  Args: {
+    showCreateBoxelSpecIntent: boolean;
+  };
+}
+
+function htmlSafeColor(color?: string) {
+  return htmlSafe(`background-color: ${color || ''}`);
+}
+
+const BoxelSpecPreviewTitle: TemplateOnlyComponent<TitleSignature> = <template>
+  Boxel Specification
+
+  <span class='has-boxel-spec' data-test-has-boxel-spec>
+    {{#if @showCreateBoxelSpecIntent}}
+      <Pill style={{htmlSafeColor 'orange'}}>No Boxel Spec</Pill>
+    {{/if}}
+  </span>
+
+  <style scoped>
+    .has-boxel-spec {
+      margin-left: auto;
+      color: var(--boxel-450);
+      font: 500 var(--boxel-font-xs);
+      letter-spacing: var(--boxel-lsp-xl);
+      text-transform: uppercase;
+    }
+  </style>
+</template>;
+
+interface ContentSignature {
+  Element: HTMLDivElement;
+  Args: {
+    showCreateBoxelSpecIntent: boolean;
+    boxelSpecInstances: CatalogEntry[];
+    selectedInstance: CatalogEntry | null;
+    createBoxelSpec: () => void;
+    selectBoxelSpec: (boxelSpec: CatalogEntry) => void;
+  };
+}
+
+const BoxelSpecPreviewContent: TemplateOnlyComponent<ContentSignature> =
+  <template>
+    <div class='boxel-spec-preview'>
+      {{#if @showCreateBoxelSpecIntent}}
+        <div class='create-boxel-spec-intent-message'>
+          Create a Boxel Specification to be able to create new instances
+        </div>
+        <BoxelButton @kind='primary' {{on 'click' @createBoxelSpec}}>
+          Create Boxel Spec
+        </BoxelButton>
+      {{else}}
+        <div class='boxel-spec-selector'>
+          <RadioInput
```

Comment:
Does it make sense to show a Radio button if there is only one spec instance?

Reply:
There can be more than one boxel spec / catalog entry since the boxel spec is simply a card that points to a code ref. 

A user can make this themselves. And I don't think we are in control of the user making an extra code spec

Reply:
I understand that there can be more than one. But I also think that in most cases there will be zero or one. So I wonder if radio input should be a special case and not the usual case.